utils/terrascript_client.py

```diff
@@ -0,0 +1,344 @@
+            self.init_values(resource, namespace_info)
+
+        tf_resources = []
+        self.init_common_outputs(tf_resources, namespace_info,
+                                 output_resource_name)
+
+        # rds instance
+        # Ref: https://www.terraform.io/docs/providers/aws/r/db_instance.html
+        try:
+            variables = self.variables[account]
+            values['db_subnet_group_name'] = variables['rds-subnet-group']
+            values['vpc_security_group_ids'] = \
+                variables['rds-security-groups'].split(',')
+        except KeyError as e:
+            logging.error("could not get an account variable: " + e.msg)
+            return
+        values['password'] = \
+            self.determine_rds_db_password(namespace_info,
+                                           output_resource_name)
+
+        tf_resource = aws_db_instance(identifier, **values)
+        tf_resources.append(tf_resource)
+        # rds outputs
+        # we want the outputs to be formed into an OpenShift Secret
+        # with the following fields
+        # db.host
+        output_name = output_resource_name + '[db.host]'
+        output_value = '${' + tf_resource.fullname + '.address}'
+        tf_resources.append(output(output_name, value=output_value))
+        # db.port
+        output_name = output_resource_name + '[db.port]'
+        output_value = '${' + tf_resource.fullname + '.port}'
+        tf_resources.append(output(output_name, value=output_value))
+        # db.name
+        output_name = output_resource_name + '[db.name]'
+        output_value = values['name']
+        tf_resources.append(output(output_name, value=output_value))
+        # db.user
+        output_name = output_resource_name + '[db.user]'
+        output_value = values['username']
+        tf_resources.append(output(output_name, value=output_value))
+        # db.password
+        output_name = output_resource_name + '[db.password]'
+        output_value = values['password']
+        tf_resources.append(output(output_name, value=output_value))
+
+        for tf_resource in tf_resources:
+            self.add_resource(account, tf_resource)
+
+    def determine_rds_db_password(self, namespace_info, output_resource_name):
+        existing_oc_resource = \
+            self.fetch_existing_oc_resource(namespace_info,
+                                            output_resource_name)
+        if existing_oc_resource is not None:
+            enc_password = existing_oc_resource['data']['db.password']
+            return base64.b64decode(enc_password)
+        return self.generate_random_password()
+
+        # TODO: except KeyError?
+        # a KeyError will indicate that this secret
+        # exists, but the db.password field is missing.
+        # this could indicate that a secret with this
+        # this name is 'taken', or that the secret
+        # was updated manually. at this point, it may
+        # be better to let the exception stop the process.
```

Comment:
I agree with not catching this except for now, this is highly unlikely if things work as they're supposed to. if it doesn't work, I want the full stack trace